modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/daemon/SelectablePeerDiscoveryDelay.scala

```diff
@@ -0,0 +1,60 @@
+package org.tessellation.sdk.infrastructure.snapshot.daemon
+
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import scala.concurrent.duration.DurationInt
+
+import org.tessellation.cli.AppEnvironment
+import org.tessellation.cli.AppEnvironment.Dev
+import org.tessellation.schema.node.NodeState.Ready
+import org.tessellation.schema.peer.Peer
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.snapshot.PeerDiscoveryDelay
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosInt
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import retry.RetryPolicies.{constantDelay, limitRetriesByCumulativeDelay}
+import retry.retryingOnFailures
+
+object SelectablePeerDiscoveryDelay {
+
+  def make[F[_]: Async](
+    clusterStorage: ClusterStorage[F],
+    appEnvironment: AppEnvironment
+  ): PeerDiscoveryDelay[F] = new PeerDiscoveryDelay[F] {
+
+    private val logger = Slf4jLogger.getLogger[F]
+
+    private val checkPeersAttemptDelay = 1.minute
+    private val checkPeersMaxDelay = 10.minutes
+    private val additionalDiscoveryDelay = 10.minutes
+
+    private val minPeersCount: PosInt = 2
+    private val fallbackPeersCount: PosInt = 1
+
+    private def waitForPeers(minPeers: PosInt): F[Set[Peer]] = retryingOnFailures[Set[Peer]](
+      limitRetriesByCumulativeDelay(
+        threshold = checkPeersMaxDelay,
+        policy = constantDelay(checkPeersAttemptDelay)
+      ),
+      a => (a.size >= minPeers).pure[F],
+      (a, details) => logger.info(s"Discovered ${a.size} selectable peers, waiting for more: $details")
+    )(clusterStorage.getResponsivePeers.map(_.filter(_.state === Ready)))
+
+    override def waitForPeers: F[Unit] = {
```

Comment:
it does not override a parent class' implementation

Reply:
Removed the `override`.